WolvenKit.App/ViewModels/Shell/AppViewModel.ComplexFiles.cs

```diff
@@ -0,0 +1,205 @@
+                if (File.Exists(xlPath) && dialogModel.IsOverwrite)
+                {
+                    File.Delete(xlPath);
+                }
+
+                var scopeName = dialogModel.SelectedBodyGender switch
+                {
+                    PhotomodeBodyGender.Cat => "photomode_cat.ent",
+                    PhotomodeBodyGender.Massive => "photomode_mm.ent",
+                    PhotomodeBodyGender.Big => "photomode_mb.ent",
+                    PhotomodeBodyGender.Male => "photomode_ma.ent",
+                    PhotomodeBodyGender.Female => "photomode_wa.ent",
+                    _ => "photomode_wa.ent",
+                };
+
+                if (!File.Exists(xlPath))
+                {
+                    var i18Node = YamlHelper.CreateLocalizationNode(jsonFileRelativePath);
+                    var scopeNode = YamlHelper.CreateScopeNode(scopeName, [relativeEntFilePath]);
+                    var rootNode = new YamlMappingNode { { "localization", i18Node }, { "resource", scopeNode } };
+                    YamlHelper.WriteYaml(xlPath, rootNode);
+                }
+            }
+
+            var absoluteAppFilePath = Path.Join(absolutePhotomodeFolder, dialogModel.SelectedApp);
+            if (dialogModel.IsCreateAppFile && File.Exists(absoluteAppFilePath))
+            {
+                var facialAnim = dialogModel.SelectedBodyGender switch
+                {
+                    PhotomodeBodyGender.Male => SelectAnimationPathViewModel.FacialAnimPathMale,
+                    PhotomodeBodyGender.Female => SelectAnimationPathViewModel.FacialAnimPathFemale,
+                    PhotomodeBodyGender.Big => SelectAnimationPathViewModel.FacialAnimPathMale,
+                    PhotomodeBodyGender.Massive => SelectAnimationPathViewModel.FacialAnimPathMale,
+                    PhotomodeBodyGender.Cat => SelectAnimationPathViewModel.FacialAnimPathFemale,
+                    _ => SelectAnimationPathViewModel.FacialAnimPathFemale,
+                };
+                DocumentTools.SetFacialAnimations(absoluteAppFilePath, facialAnim,
+                    null, SelectAnimationPathViewModel.PhotomodeAnimEntriesFemaleDefault);
+            }
+
+            _loggerService.Success("Done! Your photo mode NPV should now work!");
+        }
+        finally
+        {
+            ProjectExplorerViewModel.ResumeFileWatcherStatic();
+        }
+
+        return;
+
+        string GetRelativePath(string pathOrFileName)
+        {
+            if (activeProject.ModFiles.Contains(pathOrFileName))
+            {
+                return pathOrFileName;
+            }
+
+            if (pathOrFileName.EndsWith(".yaml"))
+            {
+                return Path.Combine(ProjectResourceHelper.AppendPersonalDirectory("r6", "tweaks"), pathOrFileName);
+            }
+
+            return Path.Combine(relativePhotoModeFolder, pathOrFileName);
```

Comment:
More general feedback: Maybe thinking of streamlining the usages of code.
Using `string.Join()`, `Path.Join()` or `Path.Combine()`?

Reply:
I copy and paste a lot, so I tend to use whatever is in the current file. But you're right.